setup.py

```diff
@@ -7,61 +7,61 @@
 from setuptools import find_packages, setup
 
 install_requires = [
-    "addfips",
-    "coloredlogs",
-    "contextily",
-    "datapackage>=1.11",
-    "catalystcoop.dbfread>=3.0",
-    "geopandas>=0.8.0",
-    "goodtables-pandas-py>=0.1.1",
-    "matplotlib",
-    "networkx>=2.2",
-    "numpy",
-    "pandas>=1.2",
-    "pyarrow>=2.0.0",
-    "pyyaml",
-    "scikit-learn>=0.20",
-    "scipy",
-    "seaborn",
-    "sqlalchemy>=1.3.0",
-    "tableschema>=1.12.3",
-    "tableschema-sql>=1.3.1",
-    "timezonefinder",
-    "tqdm",
-    "xlsxwriter",
+    "addfips>=0.3,<0.4",
+    "coloredlogs>=15.0,<16.0",
+    "contextily>=1.0,<2.0",
+    "datapackage>=1.11,<2.0",
+    "catalystcoop.dbfread>=3.0,<4.0",
+    "geopandas>=0.8.1,<0.9",
+    "goodtables-pandas-py>=0.2,<0.3",
+    "matplotlib>=3.0,<4.0",
+    "networkx>=2.2,<3.0",
+    "numpy>=1.19,<2.0",
+    "pandas>=1.2,<2.0",
+    "pyarrow>=2.0,<3.0",
+    "pyyaml>=5.0,<6.0",
+    "scikit-learn>=0.20,<1.0",
+    "scipy>=1.6,<2.0",
+    "seaborn>=0.11,<0.12",
+    "sqlalchemy>=1.3,<2.0",
+    "tableschema>=1.12.3,<2.0",
+    "tableschema-sql>=1.3.1,<2.0",
+    "timezonefinder>=5.0,<6.0",
+    "tqdm>=4.0,<5.0",
+    "xlsxwriter>=1.3,<2.0",
 ]
 
 # We are installing the PUDL module to build the docs, but the C libraries
 # required to build snappy aren"t available on RTD, so we need to exclude it
 # from the installed dependencies here, and mock it for import in docs/conf.py
 # using the autodoc_mock_imports parameter:
 if not os.getenv("READTHEDOCS"):
-    install_requires.append("python-snappy")
+    install_requires.append("python-snappy>=0.5.4,<0.6")
 
 doc_requires = [
-    "doc8",
-    "sphinx>=3.0",
-    "sphinx-issues",
-    "sphinx_rtd_theme",
+    "doc8>=0.8,<0.9",
+    "sphinx>=3.0,<4.0",
+    "sphinx-issues>=1.2,<2.0",
+    "sphinx_rtd_theme>=0.5,<0.6",
 ]
 
 test_requires = [
-    "bandit",
-    "coverage",
-    "doc8",
-    "flake8",
-    "flake8-builtins",
-    "flake8-colors",
-    "flake8-docstrings",
+    "bandit>=1.6,<2.0",
+    "coverage>=5.3,<6.0",
+    "doc8>=0.8,<0.9",
+    "flake8>=3.8,<4.0",
+    "flake8-builtins>=1.5,<2.0",
+    "flake8-colors>=0.1,<0.2",
+    "flake8-docstrings>=1.5,<2.0",
     "flake8-rst-docstrings",
```

Comment:
Missing version constraints.

Reply:
The most recent version was 0.0.14 and so I wasn't sure what to do with it! Restricting to <0.1.0 seemed a bit odd.